text_extensions_for_pandas/spanner/test_extract.py

```diff
@@ -0,0 +1,120 @@
+#
+#  Copyright (c) 2020 IBM Corp.
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+import re
+import textwrap
+
+from text_extensions_for_pandas.spanner.extract import *
+from text_extensions_for_pandas.io.spacy import make_tokens
+from text_extensions_for_pandas.io.systemt import load_dict
+from text_extensions_for_pandas.util import TestBase
+
+import spacy
+
+
+class ExtractTest(TestBase):
+
+    @staticmethod
+    def _make_tokenizer():
+        from spacy.lang.en import English
+        nlp = English()
+        return nlp.Defaults.create_tokenizer(nlp)
+
+    @staticmethod
+    def _read_file(file_name):
+        with open(file_name, "r") as f:
+            lines = [
+                line.strip() for line in f.readlines() if len(line) > 0 and line[0] != "#"
+            ]
+        return " ".join(lines)
+
+    def test_extract_dict(self):
+        file_name = "test_data/io/test_systemt/test.dict"
```

Comment:
I think the location of this file is an anachronism. Would you mind moving it to `test_data/spanner`?

Reply:
Also, I'd recommend that you remove the last three lines of the current file and replace "file_text" below with a string that exercises the major cases of dictionary extraction:
* Match at beginning of string, end of string, or in the middle of the string
* One-token match and multi-token match
* Non-match that shares the first token (and only the first token) with a two-token dictionary entry
* Two overlapping matches

Reply:
You'll also want to exercise case-insensitivity of the dictionary matching.

Reply:
> I think the location of this file is an anachronism. Would you mind moving it to test_data/spanner?

This data file is still being used in `test_systemt` also, you mean make a copy of it for spanner and replace the Haiku with something else?